lib/active_snapshot/models/snapshot_item.rb

```diff
@@ -12,22 +12,21 @@ class SnapshotItem < ActiveRecord::Base
     validates :snapshot_id, presence: true
     validates :item_id, presence: true, uniqueness: { scope: [:snapshot_id, :item_type] }
     validates :item_type, presence: true
+    validates :object, presence: true
 
     def object
       return @object if @object
 
       if ActiveSnapshot.config.storage_method_json?
-        @object = JSON.parse(self[:object])
+        @object = super() ? JSON.parse(super()) : {}
       elsif ActiveSnapshot.config.storage_method_yaml?
-        yaml_method = "unsafe_load"
+        yaml_method = YAML.respond_to?(:unsafe_load) ? :unsafe_load : :load
 
-        if !YAML.respond_to?("unsafe_load")
-          yaml_method = "load"
-        end
-
-        @object = YAML.send(yaml_method, self[:object])
+        @object = super() ? YAML.public_send(yaml_method, super()) : {}
```

Comment:
Looks like the rest of the changes in this method are no longer relevant

Reply:
Do I have to remove the rest or change the error class?